R/removal.R

```diff
@@ -206,23 +206,33 @@ removal <- function(catch,
   # some initial checks
   method <- match.arg(method)
   if (conf.level<=0 | conf.level>=1) STOP("'conf.level' must be between 0 and 1")
+  if (Tmult<1) STOP("'Tmult' should be greater than 1.")
   if (!is.vector(catch)) {
     # if a one row or column matrix then convert to a vector
-    if (is.matrix(catch) & (nrow(catch)==1 | ncol(catch)==1)) {
-      catch <- as.numeric(catch)
+    if (is.matrix(catch) & nrow(catch)==1) {
+      catch <- catch[1,]
+    } else if (is.matrix(catch) & ncol(catch)==1) {
+      catch <- catch[,1]
     } else if (is.data.frame(catch) & ncol(catch)==1) {
-      catch <- as.numeric(catch[,1])
+      catch <- catch[,1]
     } else {
       # otherwise send an error
       STOP("'catch' must be a vector.")
     }
   }
+  #Check if catch vector is not numeric
+  if (!is.numeric(catch)){
+    stop("'catch' must be a vector of numeric values.")
+  }
   if (any(is.na(catch))) {
     WARN("'NA's removed from 'catch' to continue.")
     catch <- catch[!is.na(catch)]
   }
+  #Check is.whole
+  if (!all(is.wholenumber(catch))) WARN("'catch' contains non-whole numbers.")
+  
```

Comment:
Jason ... can we move this above checking for NAs? It makes more sense to check for whole numbers after checking if it is numeric and before checking for NAs. Ultimately I don't think it changes the way the function works, but it will be more logical.

So current lines 231-233 will go before current line 227.

Reply:
is.wholenumber check moved up before checking for NAs. This caused one issue with the al() function. If NA's are present in the vector passed to is.wholenumber (e.g., 20,10,NA) a vectors with NA is returned (TRUE, TRUE, NA). The all function defaults to na.rm=FALSE. I added na.rm=TRUE in the removal file line 228 where the whole number check occurs